InventoryDistanceBasedSourceSelection/Plugin/FillSourceLatitudeAndLongitude.php

```diff
@@ -53,8 +70,10 @@ public function beforeSave(
 
                 $source->setLatitude($latLng->getLat());
                 $source->setLongitude($latLng->getLng());
-            } catch (\Exception $e) {
-                unset($e); // Silently fail geo coding
+            } catch (\Exception $exception) {
```

Comment:
I believe it is better to show only `LocalizedException` to the user, like there: https://github.com/magento/inventory/blob/c90749314415076d3df316d55de165210418f009/InventoryInStorePickupSales/Model/NotifyOrdersAreReadyForPickup.php#L119
Also, see [5.17](https://devdocs.magento.com/guides/v2.4/coding-standards/technical-guidelines.html?itm_source=devdocs&itm_medium=search_page&itm_campaign=federated_search&itm_term=Exceptions#5-exceptions).

Lets go with `$this->messageManager->addWarningMessage($exception->getMessage());` for `LocalizedException`.
For generic `\Exception` - show warning message like 'Failed to geocode the source address'.

Please don't forget to add the message to an `i18n` CSV and log the exceptions as well.